### PR TITLE
ddtrace/tracer: fix UDS/http url configuration and error messages

### PR DESCRIPTION
**Pull request by knusbaum**
This commit cleans up the configuration of the agent URL so that it reflects the actual URL being used by the Tracer. Previously it would be left as localhost:8162 if the tracer was using UDS, and the error messages would mention localhost:8126 even though it was sending to the agent over unix sockets.

In addition, it cleans up/fixes a few tests associated with this change.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Changed code has unit tests for its functionality.
- [ ] If this interacts with the agent in a new way, a system test has been added.

### CODE REVIEW

**Comment by mackjmr** [comment]
@knusbaum should we also update the agent URL when `WithUDS` is used (https://github.com/DataDog/dd-trace-go/blob/v1.44.0/ddtrace/tracer/option.go#L631-L634) ? Currently this is also left as `localhost:8162`.

File: ddtrace/tracer/option.go [hunk 1]
```diff
@@ -187,15 +188,22 @@ const maxPropagatedTagsLength = 512
...
 	c := new(config)
 	c.sampler = NewAllSampler()
-	c.agentURL = "http://" + resolveAgentAddr()
-	c.httpClient = defaultHTTPClient()
-	if url := internal.AgentURLFromEnv(); url != nil {
-		if url.Scheme == "unix" {
-			c.httpClient = udsClient(url.Path)
-		} else {
-			c.agentURL = url.String()
+	if c.agentURL == nil {
```
Comment (ajgajg1134) [inline ddtrace/tracer/option.go:191]: Won't `c.agentURL` always be nil here since the `config` has just been created?

**Review by ajgajg1134 — CHANGES_REQUESTED** [review]
Looks good, good 👀 spotting those flipped test assertions.
One question about the logic in `newConfig`

File: ddtrace/tracer/option.go (hunk 1 above)
Comment (knusbaum) [reply ddtrace/tracer/option.go]: Yes, this was actually a problem causing the issue @mackjmr noticed. I've moved it down.

**Comment by knusbaum** [comment]
Nice catch, @mackjmr. That should be fixed now.

File: ddtrace/tracer/option.go [hunk 2]
```diff
@@ -253,6 +238,21 @@ func newConfig(opts ...StartOption) *config {
...
+		c.agentURL = resolveAgentAddr()
+		if url := internal.AgentURLFromEnv(); url != nil {
+			c.agentURL = url
+		}
+	}
+	if c.agentURL.Scheme == "unix" {
+		c.httpClient = udsClient(c.agentURL.Path)
+		c.agentURL = &url.URL{
+			Scheme: "http",
```
Comment (ajgajg1134) [inline ddtrace/tracer/option.go:250]: If the agent URL scheme is `unix` should this one be `unix` here as well?

Comment (knusbaum) [reply ddtrace/tracer/option.go:250]: No. This is where we intercept `unix://` URI's and "convert" them into `http://` URIs that the http client can use.

**Review by ajgajg1134 — APPROVED** [review]

File: ddtrace/tracer/transport_test.go [hunk 3]
```diff
@@ -198,17 +198,26 @@ func TestTraceCountHeader(t *testing.T) {
 }
 
 type recordingRoundTripper struct {
-	reqs   []*http.Request
-	client *http.Client
+	reqs []*http.Request
+	//client *http.Client
```
Comment (ajgajg1134) [inline ddtrace/tracer/transport_test.go:202]: nit: do we need this commented out http.Client still?

**Review by ajgajg1134 — DISMISSED** [review]
Approved! with one optional nit

File: ddtrace/tracer/transport_test.go (hunk 3 above)
Comment (knusbaum) [reply ddtrace/tracer/transport_test.go:202]: Doh!

Comment (knusbaum) [reply ddtrace/tracer/transport_test.go:202]:
```suggestion 
```

**Review by ajgajg1134 — APPROVED** [review]

**Comment by alexstojda** [comment]
Hi! Thanks for this @knusbaum, how soon can we get this merged? :D 

@ajgajg1134

**Review by ajgajg1134 — APPROVED** [review]

**Pull request merged**